rlp/list.libevm_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package rlp
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncodeListToBuffer(t *testing.T) {
+	vals := []uint{1, 2, 3, 4, 5}
+
+	want, err := EncodeToBytes(vals)
+	require.NoErrorf(t, err, "EncodeToBytes(%T{%[1]v})", vals)
+
+	var got bytes.Buffer
+	buf := NewEncoderBuffer(&got)
```

Comment:
supernit did you know about `bytes.NewBuffer`? 😉  Same thing, but it constructs a pointer to a buffer 🤷 

```suggestion
	got := bytes.NewBuffer(nil)
	buf := NewEncoderBuffer(got)
```

Reply:
I do but only use it when I actually want to supply the buffer to avoid an allocation. I don't like call sites with opaque arguments like `nil` or booleans because they don't tell the reader anything and force them to either know or check parameters. The documentation explicitly states that the zero value is ready for use.